app/Support/SchemaInfo.php

```
@@ -0,0 +1,56 @@
+ * BookRepository::syncPublishers() and PublisherRepository::mergePublishers().
+ *
+ * The result is cached per table name for the lifetime of the request (one
+ * PHP-FPM worker process serves one request), so the `SHOW TABLES` probe runs
+ * at most once per table regardless of how many queries gate on it.
+ */
+final class SchemaInfo
+{
+    /** @var array<string, bool> */
+    private static array $tableCache = [];
+
+    public static function hasTable(\mysqli $db, string $table): bool
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $db. Configured minimum length is 3. (undefined)

(ShortVariable)